### PR TITLE
remu instruction

### PR DESCRIPTION
**Pull request by lvella**
Dismembered from PR #553.

### CODE REVIEW

**Comment by chriseth** [comment]
Should we wait for the multi-return?

**Comment by lvella** [comment]
> Should we wait for the multi-return?

should we, @georgwiese ?

**Comment by georgwiese** [comment]
Sure, #485 is done from my side!

**Comment by leonardoalt** [comment]
@lvella I think this can be updated now?

**Comment by lvella** [comment]
So, are we really using multi-return here? There will be 2 inputs and 2 outputs, meaning we have to add another assignment register.

File: riscv/src/compiler.rs [hunk 1]
```diff
@@ -1004,7 +1008,11 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
         }
         "divu" => {
             let (rd, r1, r2) = rrr(args);
-            only_if_no_write_to_zero(format!("{rd} <=Z= divu({r1}, {r2});"), rd)
+            only_if_no_write_to_zero(format!("{rd}, tmp1 <=Z,W= divremu({r1}, {r2});"), rd)
```
Comment (leonardoalt) [inline riscv/src/compiler.rs:1011]:
```suggestion
            only_if_no_write_to_zero(format!("{rd}, tmp1 <== divremu({r1}, {r2});"), rd)
```

File: riscv/src/compiler.rs [hunk 2]
```diff
@@ -1004,7 +1008,11 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
...
+        }
+        "remu" => {
+            let (rd, r1, r2) = rrr(args);
+            only_if_no_write_to_zero(format!("tmp1, {rd} <=Z,W= divremu({r1}, {r2});"), rd)
```
Comment (leonardoalt) [inline riscv/src/compiler.rs:1015]: same here

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**